spylon_kernel/scala_interpreter.py

```diff
@@ -67,17 +64,43 @@ def cleanup():
     # this from being set after SparkContext is instantiated.
     conf.conf.set("spark.repl.class.outputDir", output_dir)
 
+    # Get the application name from the spylon configuration object
+    application_name = conf.conf._conf_dict.get('spark.app.name', DEFAULT_APPLICATION_NAME)
+
+    # Force spylon to "discover" the spark python path so that we can import pyspark
+    conf._init_spark()
+
+    # Patch the pyspark.java_gateway.Popen instance to force it to pipe to the parent
+    # process so that we can catch all output from the Scala interpreter and Spark
+    # objects we're about to create
+    # Note: Making this an official part of the pyspark.java_gateway.launch_gateway
+    #
+    import pyspark.java_gateway
+    spark_jvm_proc = None
+    def Popen(*args, **kwargs):
+        """Wraps subprocess.Popen to force stdout and stderr from the child process
+        to pipe to this process without buffering.
+        """
+        nonlocal spark_jvm_proc
+        # Override these in kwargs to avoid duplicate value errors
+        kwargs['bufsize'] = 0
```

Comment:
No. Setting to zero sets the streams to unbuffered so they don't block and allow for short reads. Adding a comment with a link to the Python doc about it.